src/service/ampdoc-impl.js

```diff
@@ -476,10 +476,13 @@ export class AmpDocShadow extends AmpDoc {
  * initial configuration.
  * @param {!Window} win
  * @param {boolean} isSingleDoc
- * @return {!AmpDocService}
  */
 export function installDocService(win, isSingleDoc) {
-  return /** @type {!AmpDocService} */ (getService(win, 'ampdoc', () => {
-    return new AmpDocService(win, isSingleDoc);
-  }));
+  registerServiceBuilder(
+      win,
+      'ampdoc',
+      /* opt_constructor */ undefined,
+      () => {
+        return new AmpDocService(win, isSingleDoc);
+      });
```

Comment:
Nit: You can use the short form to fit this on one line:

`() => new AmpDocService(win, isSingleDoc));`